chia/full_node/full_node_api.py

```diff
@@ -877,21 +877,28 @@ async def declare_proof_of_space(
             new_block_gen: Optional[NewBlockGenerator]
             async with self.full_node.blockchain.priority_mutex.acquire(priority=BlockchainMutexPriority.high):
                 peak: Optional[BlockRecord] = self.full_node.blockchain.get_peak()
+                tx_peak: Optional[BlockRecord] = self.full_node.blockchain.get_tx_peak()
```

Comment:
we are getting the last transaction block a few lines down (curr_l_tb) i think we should consolidate, its under the lock so should be the same but would be cleaner

Reply:
I saw that as well. I was hesitant to mix this PR with refactoring the code below. I think it would be simpler to just call `get_tx_peak()` and use that in both places.